backend/authentication/forms.py

```diff
@@ -16,6 +16,16 @@ def get_login_url(self, request, token, next=None):
             request, token, next=next or "/#/auth/authenticated"
         )
 
+    def save(self):
+        """redefine just to transmit the login_url in test mode"""
+        email = self.cleaned_data["email"]
+        for user in self.get_users(email):
+            context = self.get_mail_context(self.request, user)
+            self.send_mail(email, context)
+            if "test" in sys.argv or "backend:test" in sys.argv:
+                # hack to pass the login url to the test suite
+                os.environ["login_url"] = context["login_url"]
```

Comment:
Wow, this is indeed an ugly hack! Environment variable to act as a global one in production code just for tests isn't a good pattern, can we imagine another strategy here? Maybe a settings? You know you can have different django settings modules for different execution environment?

Reply:
We can replace a global variable with another, anyway this code is not executed in production. Do you see a better way to pass the login_url thrown in the console to the test suite?

Reply:
I'm investigating the `locmem` email backend that should allow to watch in mails to be sent

Reply:
Now I remember there's builtin django tooling for testing email sending and contents https://docs.djangoproject.com/fr/5.0/topics/testing/tools/#django.core.mail.django.core.mail.outbox

Reply:
Yes, the `locmem` backend is automatically used in testing. I pushed a change which looks in the generated mail (I cannot look in the original context that was used to generate the mail)